src/core/agent/services/keriaNotificationService.ts

```diff
@@ -1056,6 +1010,10 @@ class KeriaNotificationService extends AgentService {
               }
             }
           }
+          await this.ipexCommunications.createLinkedIpexMessageRecord(
```

Comment:
This should be inside the `if` statement above. And why are we passing `agreeExchange` to it? It should be `grantExchange`, no?

Reply:
Currently, I see we use `agreeExchange.exn.i` as the prefix of the contact so if we passing grantExchange, we cannot update the contact in the cloud

Reply:
The contact is there in the grant too, it's the `rp` field.